Assets/Scripts/MonoBehaviours/WeaponsSystem/WeaponBehaviourScript.cs

```diff
@@ -0,0 +1,148 @@
+                {
+                    _spriteRenderer.flipX = false;
+                }
+            }
+        }
+
+        public WeaponState WeaponStateProp
+        {
+            get => weaponState;
+            set => weaponState = value;
+        }
+
+        public int CurrentMagazineAmmunition
+        {
+            get => currentMagazineAmmunition;
+            set => currentMagazineAmmunition = value;
+        }
+
+        public int CurrentTotalAmmunition
+        {
+            get => currentTotalAmmunition;
+            set => currentTotalAmmunition = value;
+        }
+
+        private void Awake()
+        {
+            _rigidbody = GetComponent<Rigidbody2D>();
+            _rigidbody.sleepMode = RigidbodySleepMode2D.StartAsleep;
+            _spriteRenderer = GetComponent<SpriteRenderer>();
+            _weaponOnGroundBehaviour = GetComponent<WeaponOnGroundBehaviour>();
+            _weaponOnGroundBehaviour.Init(weaponData);
+        
+            //TODO remove this
+            Direction = Vector2.left;
+        }
+
+        private void Update()
+        {
+            switch (weaponState)
+            {
+                case WeaponState.OnGround:
+                    _weaponOnGroundBehaviour.enabled = true;
+                    _rigidbody.WakeUp();
+                    break;
+                case WeaponState.InInventory:
+                    _weaponOnGroundBehaviour.enabled = false;
+                    _spriteRenderer.sprite = weaponData.InInventorySprite;
+                    _rigidbody.Sleep();
+                    break;
+                case WeaponState.Inactive:
+                    _weaponOnGroundBehaviour.enabled = false;
+                    _spriteRenderer.sprite = weaponData.InactiveSprite;
+                    break;
+                case WeaponState.Active:
+                    _weaponOnGroundBehaviour.enabled = false;
+                    _spriteRenderer.sprite = weaponData.ActiveSprite;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private void OnDestroy()
+        {
+            StopAllCoroutines();
+        }
+
+        public void Init(Vector2 direction)
+        {
+            this.Direction = direction;
+        }
+    
+        public void Shoot()
+        {
+            if (weaponState != WeaponState.Active)
+            {
+                Debug.Log("Tried to shoot with " + gameObject.name + " but was not active");
+                return;
+            }
+            weaponData.ShootStrategy.Shoot(this);
+        }
+
+        public void Reload()
+        {
+            if (weaponState != WeaponState.Active)
+            {
+                Debug.Log("Tried to reload with " + gameObject.name + " but was not active");
+                return;
+            }
+            weaponData.ShootStrategy.Reload(this);
+        }
+
+        public enum WeaponState
```

Comment:
Should the enum be in its own file?

Reply:
I'd say no, since the WeaponState goes alongside the Weapon's behaviour